Automattic/woocommerce-payments
# WooPayments REST: conditional initialization.

Fixes #8797
Related to https://github.com/woocommerce/woocommerce/issues/47266

#### Changes proposed in this Pull Request

Initialize REST only in a guaranteed REST context to guarantee the runtime state matches the updated state.
This shall prevent crashes due to rest controllers constructor changes during manual plugin updates.

#### Testing instructions

* Prepare a WooCommerce setup with WooPayments installed (no need for onboarding)
* Build a new zip from this branch using `npm run build:release` and manually update the plugin.
* Modify `\WC_REST_Payments_Orders_Controller::__construct` and add a new mandatory parameter
* Build a new zip from this branch using `npm run build:release` and manually update the plugin.
* After the update we should not see any mentions of the fatal error.

-------------------


**Post merge**
